extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -220,17 +220,40 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 		return err
 	}
 
+	err = wait.Poll(1*time.Second, defaults.FiveMinuteTimeout, func() (bool, error) {
+		updatedClusterResp, err := client.Management.Cluster.ByID(clusterID)
+		if err != nil {
+			return false, nil
+		}
+
+		updatedClusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane = initialControlPlaneValue
+		updatedClusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker = initialWorkerValue
+
+		_, err = client.Management.Cluster.Update(clusterResp, updatedClusterResp)
+		if err != nil {
+			return false, nil
```

Comment:
We should return the error. Would there be a reason, why wouldn't want to break the poll if the update failed?

Reply:
I can fix that, that was an oversight on my end.

Reply:
Resolved.